flang/lib/Optimizer/Builder/Character.cpp

```diff
@@ -360,6 +356,22 @@ mlir::FuncOp fir::factory::getRealloc(fir::FirOpBuilder &builder) {
                                   reallocTy);
 }
 
+mlir::FuncOp fir::factory::getLlvmStackSave(fir::FirOpBuilder &builder) {
+  auto ptrTy = builder.getRefType(builder.getIntegerType(8));
+  auto funcTy =
+      mlir::FunctionType::get(builder.getContext(), llvm::None, {ptrTy});
+  return builder.addNamedFunction(builder.getUnknownLoc(), "llvm.stacksave",
+                                  funcTy);
+}
+
+mlir::FuncOp fir::factory::getLlvmStackRestore(fir::FirOpBuilder &builder) {
+  auto ptrTy = builder.getRefType(builder.getIntegerType(8));
+  auto funcTy =
+      mlir::FunctionType::get(builder.getContext(), {ptrTy}, llvm::None);
+  return builder.addNamedFunction(builder.getUnknownLoc(), "llvm.stackrestore",
+                                  funcTy);
+}
+
```

Comment:
Is `Character.cpp` the right place for these functions?

Reply:
I suppose we could add a file for LLVM intrinsic calls or something. Is that what you were thinking?

Reply:
Yeah, this isn't dealing with a (mutable) box, character, complex, or loop.  Is `FIRBuilders.cpp` a better choice?  If not, I suppose `Character.cpp` is ok.